### PR TITLE
feat: Add Reports for teams

### LINKED ISSUE #3058: feat: UI to view team reports
Acceptance criteria:

- Admin should be able to see reports of a particular team.
- Admin should have a button to download a CSV report for the team

note: APIS will be available after #2916 

## Design

Reusing the design created for the inbox report, please modify it accordingly to team 
- sidebar nav for accessing team reports 
- filter to select specific team

Screenshot 2021-08-15 at 7 41 01 PM

### PR DESCRIPTION
**Pull request by fayazara**
## Description

Adds ability to get reports for teams.

Fixes #3058 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breakin g change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

1. Tested locally by creating teams and assigning agents to them.
2. Test cases for Vuex actions and File download helpers.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/javascript/dashboard/routes/dashboard/settings/reports/components/ReportFilters.vue [hunk 1]
```diff
@@ -185,6 +185,21 @@ export default {
...
+      if (this.type === 'label') {
+        return this.$t('LABEL_REPORTS.FILTER_DROPDOWN_LABEL');
+      }
+      if (this.type === 'inbox') {
+        return this.$t('INBOX_REPORTS.FILTER_DROPDOWN_LABEL');
+      }
+      if (this.type === 'team') {
+        return this.$t('TEAM_REPORTS.FILTER_DROPDOWN_LABEL');
+      }
+      return this.$t('INBOX_REPORTS.FILTER_DROPDOWN_LABEL');
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/reports/components/ReportFilters.vue:201]: A switch case or an object is better in this case. For eg:

```js
multiselectLabel() {
  const typeLabels = {
    agent: this.$t('AGENT_REPORTS.FILTER_DROPDOWN_LABEL'),
    label: this.$t('LABEL_REPORTS.FILTER_DROPDOWN_LABEL');
    // ..
  };
  return typeLabels[this.type] || '';
}
```

File: app/javascript/dashboard/store/modules/reports.js [hunk 2]
```diff
@@ -120,7 +120,24 @@ export const actions = {
...
+        let csvContent = 'data:text/csv;charset=utf-8,' + response.data;
+        var encodedUri = encodeURI(csvContent);
+        var downloadLink = document.createElement('a');
+        downloadLink.href = encodedUri;
+        downloadLink.download = reportObj.fileName;
+
+        document.body.appendChild(downloadLink);
+        downloadLink.click();
+        document.body.removeChild(downloadLink);
       })
```
Comment (pranavrajs) [inline app/javascript/dashboard/store/modules/reports.js:141]: This is repeated in the function above. Can you please create a common method and write a spec for it?

**Review by pranavrajs — COMMENTED** [review]
Nice work! Some minor changes required, otherwise seems fine.

File: app/javascript/dashboard/routes/dashboard/settings/reports/components/ReportFilters.vue (hunk 1 above)
Comment (fayazara) [reply app/javascript/dashboard/routes/dashboard/settings/reports/components/ReportFilters.vue]: Resolved in https://github.com/chatwoot/chatwoot/pull/3116/commits/ac0c18a200d56fea549ababc00af74991f937120

File: app/javascript/dashboard/store/modules/reports.js (hunk 2 above)
Comment (fayazara) [reply app/javascript/dashboard/store/modules/reports.js]: Resolved in https://github.com/chatwoot/chatwoot/pull/3116/commits/2e9641d9a5ee2d671424162eae37474b7aa4a6c6

**Review by muhsin-k — APPROVED** [review]

**Review by pranavrajs — APPROVED** [review]

**Pull request merged**